src/gui/wizard/owncloudadvancedsetuppage.cpp

```diff
@@ -344,6 +411,17 @@ void OwncloudAdvancedSetupPage::slotSelectFolder()
     setErrorString(errorStr);
 }
 
+
+void OwncloudAdvancedSetupPage::setLocalFolderPushButtonPath(const QString &path)
+{
+    auto prettyPath = path;
+
+    const QString homeDir = QDir::homePath() + QString("/");
+    prettyPath.replace(prettyPath.indexOf(homeDir), homeDir.size(), "");
```

Comment:
You could avoid introducing temporary non-const variable if using **QString ::chopped**. Also, please double check if you can safely assume **prettyPath.indexOf(homeDir)** is never -1. Otherwise, you could store that index in the separate const variable and then use it with **chopped** only if it's **>= 0**.

Reply:
If I use the `chopped()` approach then I need to check if `prettyPath` begins with `homeDir`? This would add an extra step.

Reply:
Note that today you should check as well... `indexOf(homeDir)` might not be 0 if you're unlucky and in the middle of prettyPath there is a string which looks like homeDir. Using replace + indexOf blindly for things where you assume it should starts with a substring sounds like a bug waiting to happen.

I'd say use replace less and higher level more expressive constructs when they are available. ;-)

Maybe a bad example but honestly I *very* seldomly use replace with QString, except if it's for an actual search & replace (possibly with a regex) type of operation.